content/blog/model-calibration/index.Rmd

```diff
@@ -0,0 +1,409 @@
+outcome, has the value of "tails".]  that match such probability. 
+
+For example, if we collect a group of the predictions whose probabilities are 
+estimated to be about 10%. We should expect that about 10% of the those in the 
+group to indeed be events. The plots shown below can be used as diagnostics to 
+see if our predictions are consistent with the observed event rates. 
+
+
+### Example Data
+
+If you would like to follow along, load the probably and dplyr packages 
+into you R session.
+
+```{r}
+library(tidymodels)
+library(probably)
+```
+
+The probably package comes with a few data sets. For most of the examples in this post,
+we will use `segment_logistic`. It is an example data set that contains
+predictions, and their probabilities. `Class` contains the outcome of 
+"good" and "poor", `.pred_good` contains the probability that the event is "good". 
+
+```{r}
+segment_logistic
+```
+
+
+### Binned Plot
+
+On smaller data sets, it is a challenging to obtain an accurate *event rate* 
+for a given probability. For example, if there are 5 predictions with about a 50% 
+probability, and 3 of those are events, the plot would show a 60% event rate.
+This comparison would not be appropriate because there are not enough predictions
+to really determine how close to 50% the model really is. 
+
+The most common approach to group the probabilities
+into bins, or buckets.  Usually, the data is split into 10 discrete buckets,
+from 0 to 1 (0 - 100%). The *event rate* and the *bin midpoint* is 
+calculated for each bin. 
+
+In probably, binned calibration plots can be created using `cal_plot_breaks()`. 
+It expects a data set, and the un-quoted variable names that contains the events
+(`truth`), and the probabilities (`estimate`). For the example here, we pass the 
+`segment_logistic` data set, and use `Class` and `.pred_good` as the arguments. 
+By default, this function will create a calibration plot with 10 buckets (breaks):
+
+```{r, fig.alt = "Calibration plot with 10 bins, created with the cal_plot_breaks() function"}
+segment_logistic %>% 
+  cal_plot_breaks(Class, .pred_good)
+```
+
+```{r}
+#| include: false
+bin_plot_data <- segment_logistic %>% .cal_binary_table_breaks(Class, .pred_good)
+```
+
+The calibration plot for the ideal model will essentially be perfect incline line
+that start at (0,0) and ends in (1,1). In the case of this model, we can see 
+that the seventh point has an event rate of 
+`r round(bin_plot_data$event_rate[7] * 100, 1)`% despite having estimated 
+probabilities ranging from 60% to 70%. This indicates that the model is not creating
+predictions in this region that are consistent with the data (i.e., it is 
+under-predicting). 
+
+The number of bins in `cal_plot_breaks()` can be adjusted using `num_breaks`. 
+Here is an example of what the plot looks like if we reduce the bins from 10,
+to 5:
+
+```{r, fig.alt = "Calibration plot with 5 bins, created with the cal_plot_breaks() function"}
```

Comment:
```suggestion
```{r, fig.alt = "A calibration like that above, but with half as many bins. In this version of the plot, the solid line is less jagged, though still shows that dots consistently lie below the dashed line beyond a predicted probability of 0.5."}
```

Once you've described visual elements of a plot, feel free to reference them in later alt text ^^